tests/languages/pygrep_test.py

```diff
@@ -38,3 +38,24 @@ def test_ignore_case(some_files, cap_out):
     out = cap_out.get()
     assert ret == 1
     assert out == 'f2:1:[INFO] hi\n'
+
+
+def test_null_data(some_files, cap_out):
+    ret = pygrep.main(('--null-data', r'foo\nbar', 'f1', 'f2', 'f3'))
+    out = cap_out.get()
+    assert ret == 1
+    assert out == 'f1:0-7:foo\nbar\n'
+
+
+def test_null_data_dotall_flag_is_enabled(some_files, cap_out):
+    ret = pygrep.main(('--null-data', r'o.*bar', 'f1', 'f2', 'f3'))
+    out = cap_out.get()
+    assert ret == 1
+    assert out == 'f1:1-7:oo\nbar\n'
```

Comment:
I think my ideal output for this would show the full line numbers, the character offsets are a bit misleading (and less useful) imo

Reply:
that said, as a v1, I'd be ok with the same output that `grep -z` gives

Reply:
I'm not a big fan of the `grep -z` output in general, I always use it in conjuction with `-bo` to get only the matching part and the byte offset instead of the line number (which is always `1` when you don't have null bytes in the file).